### PR TITLE
fix: make most of the addresses to be kopiable

### PR DESCRIPTION
**Pull request by NeOMakinG**
## Description

I did try to find every places where we display addresses with ellipsis, and add a copy button so our users can copy the full address.

Take special care to the AccountDropdown component which are the biggest changes

## Issue (if applicable)

partially #7624

## Risk
Low
> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

## Testing
Where I added a copy component:
- Claim cosmos
- Cosmos claim status
- Receive
- Send
- Swapper step
- Transaction history addresses
- FoxFarming claim and claim status (FoxFarming is crash on claim?)
- Foxy claim and claim status (I don't have any fox in the foxy opportunity...)
- AccountDropdown
All those places now has a copy button to copy the full address, go through them and verify that the button works properly (the copy contain the full address)

### Engineering
n/a

### Operations
n/a
- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

## Screenshots (if applicable)
Claim cosmos
image
Cosmos claim status
image
Receive
image
Send
image
Swapper step
image
Transaction history addresses
image
FoxFarming claim and claim status (FoxFarming is crash on claim?)
Foxy claim and claim status (I don't have any fox in the foxy opportunity...)

AccountDropdown
image
image

### CODE REVIEW

**Comment by 0xean** [comment]
best PR title ever

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Tested locally, while this LGTM visually and high-level from a functional point of view, major concerns here re: `<AccountDropdown />` impl.

- UTXO accounts dropdown copies the xpub of the first scriptType 🚫 

image

- Address-selection is impossible when `<AccountDropdown />` is disabled

image

Not sure whether that's a feature or a bug @NeOMakinG - almost feels like this guy should allow copying?

image

- Address-based chains account dropdown copies the pubkey (❓)

image

Question mark here, see inline comment, we should probably add the copy button in the individual option vs. the group option for consistency here too

- NFT Modal ✅ 

image

- Send Modal 🚫 

Same issue as above for account-based chains, xpub is copied for UTXOs

image

- Swapper Step ✅ 

Confirmed the *receive* address (i.e not the pubkey for account-based chains) is copied

image
 
- Tx history address rows ✅ 
 
Confirmed the correct address is copied here for EVMs

image

And for UTXOs:

image

- Cosmos SDK staking ✅ 

image
image
image


- FOX Farming Claim confirm/status 

Unable to test locally, but conceptually looks sane

- FOXy claim status

No claims available but looks sane too!

- WC dApps session proposal modal 🚫 

image

- WC dApps active session dropdown ✅ 

image

Not related to this PR, but we should either remove duplicates, or add the network name/icon to the row, as this looks odd currently

File: src/components/AccountDropdown/AccountDropdown.tsx [hunk 1]
```diff
@@ -163,6 +164,7 @@ const MenuOptions = ({
             <AccountSegment
               title={translate('accounts.accountNumber', { accountNumber })}
               subtitle={subtitle}
+              address={fromAccountId(firstAccountId).account}
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply src/components/AccountDropdown/AccountDropdown.tsx]: Removed for UTXOs in [a4488a6](https://github.com/shapeshift/web/pull/7663/commits/a4488a694785a80349acd54c1ca467782b1a8ee7)

File: src/components/AccountDropdown/AccountDropdown.tsx [hunk 2]
```diff
@@ -315,6 +317,30 @@ export const AccountDropdown: FC<AccountDropdownProps> = memo(
     if (!isValidAccountNumber(accountNumber)) return null
     if (!Object.keys(accountIdsByNumberAndType).length) return null
     if (!accountLabel) return null
+    if (!selectedAccountId) return null
+
+    if (isDropdownDisabled) {
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply src/components/AccountDropdown/AccountDropdown.tsx:323]: Safer home in [a4488a6](https://github.com/shapeshift/web/pull/7663/commits/a4488a694785a80349acd54c1ca467782b1a8ee7) and removed from the `Send from` for UTXos

File: src/components/Modals/Send/views/Confirm.tsx
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply src/components/Modals/Send/views/Confirm.tsx:1]: Done in [a4488a6](https://github.com/shapeshift/web/pull/7663/commits/a4488a694785a80349acd54c1ca467782b1a8ee7)

File: src/components/Modals/Send/views/Confirm.tsx [hunk 3]
```diff
@@ -155,7 +157,11 @@ export const Confirm = () => {
             <Row.Label>
               <Text translation={'modals.send.confirm.sendTo'} />
             </Row.Label>
-            <Row.Value>{vanityAddress ? vanityAddress : <MiddleEllipsis value={to} />}</Row.Value>
+            <Row.Value>
+              <InlineCopyButton value={isNonEmptyString(vanityAddress) ? vanityAddress : to}>
+                {vanityAddress ? vanityAddress : <MiddleEllipsis value={to} />}
+              </InlineCopyButton>
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply src/components/Modals/Send/views/Confirm.tsx:163]: Happy with both, let's ask the product team

File: src/components/TransactionHistoryRows/TransactionDetails/Address.tsx [hunk 4]
```diff
@@ -18,24 +19,28 @@ export const Address = ({
...
-      p={0}
-      onClick={handleClick}
-      height='auto'
-      fontFamily='monospace'
-      _hover={buttonHover}
-      fontSize='inherit'
-    >
-      <MiddleEllipsis value={ens || address} />
-    </Button>
+    <InlineCopyButton value={ens ?? address}>
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply src/components/TransactionHistoryRows/TransactionDetails/Address.tsx:22]: Happy with both, let's ask the product team

**Comment by NeOMakinG** [comment]
@gomesalexandre for the copy button inside the wallet window, I like the idea but I feel like it's not a good home without some extra work:

If the wallet supports multiple chains, I would expect to be able to find the addresses for all of the chains, or which rule should we follow? 

We might want to keep this stuff for a follow up with some extra product work on it

All over points have been fixed in [a4488a6](https://github.com/shapeshift/web/pull/7663/commits/a4488a694785a80349acd54c1ca467782b1a8ee7)

File: src/components/AccountDropdown/AccountChildOption.tsx [hunk 5]
```diff
@@ -16,18 +18,27 @@ type AccountChildRowProps = {
   cryptoBalance: string
   symbol: string
   onOptionClick: (accountId: AccountId) => void
+  chainId: ChainId
```
Comment (gomesalexandre) [inline src/components/AccountDropdown/AccountChildOption.tsx:21]: @NeOMakinG do we need this? I.e `fromAccountId(accountId).chainId` will yield the `ChainId`  (or alternatively, no need to explicitly check for `isUtxoChainId()`, you can even use `isUtxoAccountId()`)

File: src/components/AccountDropdown/AccountDropdown.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/components/AccountDropdown/AccountDropdown.tsx:323]: dis still present in `<AccountDropdown />`?

File: src/components/Modals/Send/views/Confirm.tsx
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/components/Modals/Send/views/Confirm.tsx:1]: `<AccountDropdown />` logic still present

File: src/components/AccountDropdown/AccountChildOption.tsx (hunk 5 above)
Comment (NeOMakinG) [reply src/components/AccountDropdown/AccountChildOption.tsx]: IMO we don'tm I used `fromAccountId` inside another place and it's 100% the right thing to do, changed in [5143ec3](https://github.com/shapeshift/web/pull/7663/commits/5143ec32b38ab37e8d2a9ad4e93d2838deaa9c01)

File: src/components/Modals/Send/views/Confirm.tsx
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply src/components/Modals/Send/views/Confirm.tsx:1]: Removed in [5143ec3](https://github.com/shapeshift/web/pull/7663/commits/5143ec32b38ab37e8d2a9ad4e93d2838deaa9c01), let's choose every places we need this in a follow up PR prior research of these places

File: src/components/AccountDropdown/AccountDropdown.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply src/components/AccountDropdown/AccountDropdown.tsx:320]: removed in [5143ec3](https://github.com/shapeshift/web/pull/7663/commits/5143ec32b38ab37e8d2a9ad4e93d2838deaa9c01)

**Review by gomesalexandre — COMMENTED** [review]
Tested locally:

### `<AccountDropdown />` copy button is disabled for UTXOs

image

### `<AccountDropdown />` copy button is happy for non-UTXOs (ish)

- Native

image

- MM

image

- This rugs swapper styles (see against develop on the right)

image

- As mentioned previously, I'm not sure how I feel about this - `<AccountDropdown />` doesn't really generalize. There are a bunch of places where this just feels weird (e.g send, where you have the copy button at confirm time, and receive, where you already have a receive button), this requires an additional click to see it, and most importantly, this doesn't work for most users without multiple accounts for said chains, since there is no copy button then. I'd really be in favour of hunting individual usages some more and having line items instead of the seemingly more holistic, but seemingly broken current implementation.
Send input and confirm in your latest diff is a great example of a copy button which looks sane:

image
image

- Swapper step receive addy is happy

image

- Copy button is missing in DeFi `<Approve />`, feels like this would be a good contender?

image

- WC dApps session proposal line items are happy

image

- WC dApps overview copy button is happy

image

- Fiat ramps buy button is still happy

image

- Fiat ramps sell button is still happy

image

@NeOMakinG not introduced by your PR but noticed it while testing it, the from address doesn't really apply for UTXOs, can we remove the row entirely in a follow-up? That is absolutely wrong currently🙏🏽 

- Asset Page

image

Also an improvement for a follow-up, looks like a very good contender here too... although we don't show addies. Feels like we could kill two birds one stone and both add ellipsises here for non-UTXOs, the same we do in wallet page, as well as adding a copy button. See wallet page for ref:

image

File: src/components/AccountDropdown/AccountChildOption.tsx [hunk 6]
```diff
@@ -5,7 +5,7 @@ import {
   Stack,
   useColorModeValue,
 } from '@chakra-ui/react'
-import type { AccountId } from '@shapeshiftoss/caip'
+import { type AccountId } from '@shapeshiftoss/caip'
```
Comment (gomesalexandre) [inline src/components/AccountDropdown/AccountChildOption.tsx:8]: rever type-only import

File: src/components/AccountDropdown/AccountDropdown.tsx [hunk 7]
```diff
@@ -315,6 +315,7 @@ export const AccountDropdown: FC<AccountDropdownProps> = memo(
     if (!isValidAccountNumber(accountNumber)) return null
     if (!Object.keys(accountIdsByNumberAndType).length) return null
     if (!accountLabel) return null
+    if (!selectedAccountId) return null
```
Comment (gomesalexandre) [inline src/components/AccountDropdown/AccountDropdown.tsx:318]: q: Do we need this as part of this PR? `<AccountDropdown />` is inherently bug-prone and unless we explicitly need this, I'd revert it

**Review by gomesalexandre — COMMENTED** [review]
Reretest:

- Swapper styles are unborked for MM

image

- Swapper styles are still happy with native

image

- "Send To" line item is happy, but "From" copy button is gone from send confirm 🚫 

image

- wc session proposal buttons are still happy

image

- wc overview buttons are still happy

image

File: src/components/AccountDropdown/AccountChildOption.tsx (hunk 6 above)
Comment (NeOMakinG) [reply src/components/AccountDropdown/AccountChildOption.tsx]: Reverted in [d831295](https://github.com/shapeshift/web/pull/7663/commits/d83129520fef459ee13122165049b64ef0bc470d)

File: src/components/AccountDropdown/AccountDropdown.tsx (hunk 7 above)
Comment (NeOMakinG) [reply src/components/AccountDropdown/AccountDropdown.tsx]: Nop we don't, removed in [d831295](https://github.com/shapeshift/web/pull/7663/commits/d83129520fef459ee13122165049b64ef0bc470d)

**Review by gomesalexandre — APPROVED** [review]

File: src/components/TransactionHistoryRows/TransactionDetails/Address.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (reallybeard) [reply src/components/TransactionHistoryRows/TransactionDetails/Address.tsx:22]: Copy the actual address here as well.

File: src/components/Modals/Send/views/Confirm.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply src/components/Modals/Send/views/Confirm.tsx]: Done in [892ba4f](https://github.com/shapeshift/web/pull/7663/commits/892ba4f608d1f83396523f278aed10cd8fbc7735)

File: src/components/TransactionHistoryRows/TransactionDetails/Address.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply src/components/TransactionHistoryRows/TransactionDetails/Address.tsx]: Done in [892ba4f](https://github.com/shapeshift/web/pull/7663/commits/892ba4f608d1f83396523f278aed10cd8fbc7735)

**Pull request merged**